Mathlib/Order/OmegaCompletePartialOrder.lean

```diff
@@ -268,6 +287,21 @@ def Continuous' (f : α → β) : Prop :=
   ∃ hf : Monotone f, Continuous ⟨f, hf⟩
 #align omega_complete_partial_order.continuous' OmegaCompletePartialOrder.Continuous'
 
+lemma IsLUB_of_ScottContinuous {c : Chain α} {f : α → β} (hf : ScottContinuous f) :
+    IsLUB (Set.range (Chain.map c ⟨f, (ScottContinuous.monotone hf)⟩)) (f (ωSup c)) := by
+  simp only [map_coe, OrderHom.coe_mk]
+  rw [(Set.range_comp f ↑c)]
+  apply hf
+  exact Set.range_nonempty ↑c
+  exact IsChain.directedOn (isChain_range c)
+  exact IsLUB_range_ωSup c
```

Comment:
Could you please format this with subgoals `·` or move them as inline arguments in the place where the corresponding goal is created?

Reply:
Moved inline.